### PR TITLE
[BACKLOG-36323] - changes for searching inside sub folders

### PR DESCRIPTION
**Pull request by sVarma1992**
[BACKLOG-36323] - changes for searching inside sub folders

### CODE REVIEW

**Comment by buildguy** [comment]
### :warning: Build finished in 5m 27s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
plugins/file-open-save-new/api,plugins/file-open-save-new/core
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 3 test(s) failed:

<details><summary><b>org.pentaho.di.plugins.fileopensave.controllers.FileControllerTest.initializationError</b> (click to expand)</summary><p>

```
org/hamcrest/SelfDescribing
```

</p></details>

<details><summary><b>org.pentaho.di.plugins.fileopensave.controllers.recent.RecentUtilsTest.initializationError</b> (click to expand)</summary><p>

```
org/hamcrest/SelfDescribing
```

</p></details>

<details><summary><b>org.pentaho.di.plugins.fileopensave.providers.vfs.model.VFSFileTest.initializationError</b> (click to expand)</summary><p>

```
org/hamcrest/SelfDescribing
```

</p></details>

  
Tests run: **3**, Failures: **3**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi-11/job/PR-8645/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/local/LocalFileProvider.java [hunk 1]
```diff
@@ -114,6 +115,28 @@ public class LocalFileProvider extends BaseFileProvider<LocalFile> {
     return getFiles( file, filters );
   }
 
+  @Override
+  public List<LocalFile> searchFiles( LocalFile file, String filters, String searchString, VariableSpace space )
+    throws FileException {
+    List<LocalFile> files = new ArrayList<>();
+    try ( Stream<Path> pathStream = Files.walk( Paths.get( file.getPath() ) ) ) {
+      pathStream.forEach( path -> {
+        String name = path.getFileName().toString();
```
Comment (peterrinehart) [inline plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/local/LocalFileProvider.java:124]: This line throws a null pointer exception on Linux when trying to search from the root directory /

**Review by peterrinehart — CHANGES_REQUESTED** [review]
See comment; this mostly works but need to check the edge case where you are searching from the root of the local file system on Linux.

**Comment by buildguy** [comment]
### :warning: Build finished in 6m 34s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
plugins/file-open-save-new/api,plugins/file-open-save-new/core
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 3 test(s) failed:

<details><summary><b>org.pentaho.di.plugins.fileopensave.controllers.FileControllerTest.initializationError</b> (click to expand)</summary><p>

```
org/hamcrest/SelfDescribing
```

</p></details>

<details><summary><b>org.pentaho.di.plugins.fileopensave.controllers.recent.RecentUtilsTest.initializationError</b> (click to expand)</summary><p>

```
org/hamcrest/SelfDescribing
```

</p></details>

<details><summary><b>org.pentaho.di.plugins.fileopensave.providers.vfs.model.VFSFileTest.initializationError</b> (click to expand)</summary><p>

```
org/hamcrest/SelfDescribing
```

</p></details>

  
Tests run: **3**, Failures: **3**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi-11/job/PR-8645/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :warning: Build finished in 3m 27s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
plugins/file-open-save-new/api,plugins/file-open-save-new/core
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 3 test(s) failed:

<details><summary><b>org.pentaho.di.plugins.fileopensave.controllers.FileControllerTest.initializationError</b> (click to expand)</summary><p>

```
org/hamcrest/SelfDescribing
```

</p></details>

<details><summary><b>org.pentaho.di.plugins.fileopensave.controllers.recent.RecentUtilsTest.initializationError</b> (click to expand)</summary><p>

```
org/hamcrest/SelfDescribing
```

</p></details>

<details><summary><b>org.pentaho.di.plugins.fileopensave.providers.vfs.model.VFSFileTest.initializationError</b> (click to expand)</summary><p>

```
org/hamcrest/SelfDescribing
```

</p></details>

  
Tests run: **3**, Failures: **3**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi-11/job/PR-8645/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/local/LocalFileProvider.java (hunk 1 above)
Comment (sVarma1992) [reply plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/local/LocalFileProvider.java]: Solved the issue and pushed the changes

**Pull request merged**